subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -549,17 +549,22 @@ struct ztest_unit_test *z_ztest_get_next_test(const char *suite, struct ztest_un
 #ifdef CONFIG_ZTEST_SHUFFLE
 static void z_ztest_shuffle(void *dest[], intptr_t start, size_t num_items, size_t element_size)
 {
+	void *tmp;
+
+	/* Initialize dest array */
 	for (size_t i = 0; i < num_items; ++i) {
-		int pos = sys_rand32_get() % num_items;
-		const int start_pos = pos;
+		dest[i] = (void *)(start + (i * element_size));
+	}
 
-		/* Get the next valid position */
-		while (dest[pos] != NULL) {
-			pos = (pos + 1) % num_items;
-			__ASSERT_NO_MSG(pos != start_pos);
```

Comment:
Do we not care about what this is asserting anymore?

Reply:
This assert was to catch an unlikely scenario of getting stuck in an infinite loop where `dest` has no NULL pointers.